protocol/communities/manager_test.go

```diff
@@ -201,7 +201,14 @@ func (s *ManagerSuite) TestStartHistoryArchiveTasksInterval() {
 	// Due to async exec we need to wait a bit until we check
 	// the task count.
 	time.Sleep(5 * time.Second)
-	s.Require().Len(s.manager.historyArchiveTasks, 1)
+
+	// sync.Map doesn't have a Len function, so we need to count manually
+	count := 0
+	s.manager.historyArchiveTasks.Range(func(_, _ interface{}) bool {
+		count++
+		return true
+	})
+	s.Require().Equal(count, 1)
```

Comment:
It seems that introducing a helper function here could help us reduce code duplication